sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java

```diff
@@ -391,34 +397,34 @@ String getContentType() {
      * batch operation into the overall request and then returns nothing as the response.
      */
     private Mono<HttpResponse> setupBatchOperation(HttpRequest request) {
-        int contentId = Integer.parseInt(request.getHeaders().remove(CONTENT_ID).getValue());
-
-        StringBuilder batchRequestBuilder = new StringBuilder();
-        appendWithNewline(batchRequestBuilder, "--" + batchBoundary);
-        appendWithNewline(batchRequestBuilder, BATCH_OPERATION_CONTENT_TYPE);
-        appendWithNewline(batchRequestBuilder, BATCH_OPERATION_CONTENT_TRANSFER_ENCODING);
-        appendWithNewline(batchRequestBuilder, String.format(BATCH_OPERATION_CONTENT_ID_TEMPLATE, contentId));
-        batchRequestBuilder.append(BlobBatchHelper.HTTP_NEWLINE);
-
-        String method = request.getHttpMethod().toString();
-        String urlPath = request.getUrl().getPath();
-        String urlQuery = request.getUrl().getQuery();
-        if (!ImplUtils.isNullOrEmpty(urlQuery)) {
-            urlPath = urlPath + "?" + urlQuery;
-        }
-        appendWithNewline(batchRequestBuilder, String.format(OPERATION_TEMPLATE, method, urlPath, HTTP_VERSION));
-
-        request.getHeaders().stream()
-            .filter(header -> !X_MS_VERSION.equalsIgnoreCase(header.getName()))
-            .forEach(header -> appendWithNewline(batchRequestBuilder,
-                String.format(HEADER_TEMPLATE, header.getName(), header.getValue())));
+        return Mono.fromRunnable(() -> {
+            int contentId = Integer.parseInt(request.getHeaders().remove(CONTENT_ID).getValue());
+
+            StringBuilder batchRequestBuilder = new StringBuilder();
+            appendWithNewline(batchRequestBuilder, "--" + batchBoundary);
+            appendWithNewline(batchRequestBuilder, BATCH_OPERATION_CONTENT_TYPE);
+            appendWithNewline(batchRequestBuilder, BATCH_OPERATION_CONTENT_TRANSFER_ENCODING);
+            appendWithNewline(batchRequestBuilder, String.format(BATCH_OPERATION_CONTENT_ID_TEMPLATE, contentId));
+            batchRequestBuilder.append(BlobBatchHelper.HTTP_NEWLINE);
+
+            String method = request.getHttpMethod().toString();
+            String urlPath = request.getUrl().getPath();
+            String urlQuery = request.getUrl().getQuery();
+            if (!ImplUtils.isNullOrEmpty(urlQuery)) {
+                urlPath = urlPath + "?" + urlQuery;
```

Comment:
nit: `urlPath += "?" + urlQuery;` ?

Reply:
I will add an issue to change this next week as it won't touch API

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/5942